design/aws-ecs-scheduled-ecs-task-construct.md

```diff
@@ -0,0 +1,141 @@
+
+This would mean adding a new ECS CDK construct `ScheduledEc2Task` and properties `ScheduledEc2TaskProps`, that would take in the necessary properties required to create a Task Definition, an EventRuleTarget as well as an AWS EventRule. To allow for flexibility, container overrides will also be supported properties.
+
+Note: Currently, there is only [CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-reference-ecs.html) support for scheduled tasks on EC2, and not on Fargate. CloudFormation currently only supports clusters, services and taskdefinitions: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-reference-ecs.html.
+
+TaskDefinitions contain a network mode, and for EC2, we can use none, bridge or the many others. To support Fargate, we would need to use a awsvpc as the NetworkMode of the TaskDefinition, but this requires us to create a NetworkConfiguration, which is not supported in the TaskDefinition, but is supported in the Service.
+
+For this construct, we're creating a scheduled task (that is standalone task, not backed with a service), and for Fargate to support this feature, we'd need to have CloudFormation create the concept of a Standalone Task/RunTask (not Task Definition) that can take a NetworkConfiguration as a property. It works for ECS because we're using bridge as the NetworkMode, which does not require a NetworkConfiguration to be present.
+
+## General approach
+
+The new [`ecs.ScheduledEc2Task`] class will include an L3 construct for:
+
+* ScheduledEc2Task
+
+A `ScheduledEc2Task` will create a task definition with the specified container. An `Ec2EventRuleTarget` will be created and any container overrides will be applied to an AWS `EventRule` (indicating how frequently the task should be run). Based on the `EventRule` schedule, a task will run on the EC2 instances specified in the cluster. 
+
+The following container overrides can be passed to the container to enable flexibility:
+
+* commands
+* cpu
+* environment variables
+* resource requirements (currently only a resource type of GPU is supported)
+
+## Code changes
+
+Given the above, we should make the following changes to support scheduled tasks on ECS:
+1. Create `ScheduledEc2TaskProps` interface  and `ScheduledEc2Task` construct
+
+# Part 1: Create `ScheduledEc2TaskProps` interface  and `ScheduledEc2Task` construct  
+
+The `ScheduledEc2TaskProps` interface will contain properties to construct the Ec2TaskDefinition, Ec2EventRuleTarget and EventRule, as well as any container overrides that are optional:
+
+```ts
+export interface ScheduledEc2TaskProps {
+  /**
+   * The cluster where your service will be deployed.
+   */
+  readonly cluster: ICluster;
+
+  /**
+   * The image to start.
+   */
+  readonly image: ContainerImage;
```

Comment:
@SoManyHs Where did we land on exposing task def parameters up through the services for L3s?  Am I locked out of L3s if I want to use multiple containers?

Reply:
+1 re: addressing the multiple containers question

Reply:
Single container tasks represent over 90% for both ECS & Fargate, so this is probably fine :)

Reply:
I'm of the opinion that we should optionally allow people to pass a task definition into the L3 construct, not just a container image. I don't see any reason why we couldn't allow both

Reply:
I believe this was discussed previously, and the decision was to not accept a taskdef as input for this construct. @clareliguori